### PR TITLE
Removing Enzyme and updating test files to use testing-library

### PR DESCRIPTION
**Pull request by ribeiroguilherme**
## 📋 Pull Request Checklist

- [ ] I have added unit tests to cover my changes.
- [ ] I have added or updated Storybook stories where applicable.
- [ ] I have tested the changes manually in the local environment.
- [ ] I have checked that no PII data is being sent on analytics events
- [ ] All E2E tests are passing, and I have added new tests if necessary.
- [ ] All interfaces and types introduced or updated are strictly typed. 

---

## 📝 Summary
- Removed enzyme
- Removed package resolution needed for enzyme
- Refactored tests to use testing-library
- Added one testing guideline to aim for tests without side-effects

---

## 🧪 Tested scenarios

---

## 🔗 Related GitHub Issue / Internal Ticket number

**Closes**:  

---

### CODE REVIEW

File: packages/lib/src/components/Card/components/CardInput/CardInput.test.tsx [hunk 1]
```diff
@@ -32,25 +30,31 @@ const cardInputRequiredProps = {
...
-        const wrapper = getWrapper(<CardInput {...cardInputRequiredProps} />);
-        expect(wrapper.find('[data-cse="encryptedCardNumber"]')).toHaveLength(1);
-        expect(wrapper.find('[data-cse="encryptedExpiryDate"]')).toHaveLength(1);
-        expect(wrapper.find('[data-cse="encryptedSecurityCode"]')).toHaveLength(1);
+        const { container } = renderCardInput(<CardInput {...cardInputRequiredProps} />);
+        /* eslint-disable testing-library/no-node-access, testing-library/no-container */
+        expect(container.querySelector('[data-cse="encryptedCardNumber"]')).toBeTruthy();
+        expect(container.querySelector('[data-cse="encryptedExpiryDate"]')).toBeTruthy();
+        expect(container.querySelector('[data-cse="encryptedSecurityCode"]')).toBeTruthy();
+        /* eslint-enable testing-library/no-node-access, testing-library/no-container */
```
[the remark this answers is not in the document]
Comment (ribeiroguilherme) [reply packages/lib/src/components/Card/components/CardInput/CardInput.test.tsx:50]: I pushed some improvements, but it is just a big refactor for this test file to have it all fixed. This should be tackled in another task and not as part of this one.

File: .windsurf/rules/code-guide.md [hunk 2]
```diff
@@ -471,6 +471,7 @@ Jest 29 + `@testing-library/preact` + `@testing-library/jest-dom`
 - **DO** use `waitFor()` for async assertions
 - **DO** test keyboard navigation and ARIA attributes
 - **DO** test error states and edge cases
+- **DO** maintain strict test isolation. Every test must be independent and idempotent, ensuring that the execution of one test does not influence the outcome of another through shared memory or persistent state.
```
Comment (ScottiBR) [inline .windsurf/rules/code-guide.md:474]: nice!

File: packages/lib/src/components/internal/SecuredFields/SFP/SecuredFieldsProvider.test.tsx [hunk 3]
```diff
@@ -86,26 +82,37 @@ const mockCSF = {
...
-        wrapper.instance().handleOnConfigSuccess();
-        expect(wrapper.instance().state.status).toBe('ready');
+    test('Loading state', async () => {
+        nodeHolder.innerHTML = mockNode;
+        renderSFP();
+
+        expect(sfp.state.status).toBe('loading');
+        await act(() => {
+            sfp.handleOnConfigSuccess();
+        });
```
Comment (ScottiBR) [inline packages/lib/src/components/internal/SecuredFields/SFP/SecuredFieldsProvider.test.tsx:114]: correct pattern

File: packages/lib/src/components/Card/components/CardInput/CardInput.dualBrandUI.test.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        expect(container.querySelector('[data-cse="encryptedSecurityCode"]')).toBeTruthy();
+        /* eslint-enable testing-library/no-node-access, testing-library/no-container */
+
+        expect(screen.queryByText('Card Brand')).not.toBeInTheDocument();
+    });
+
+    test('Renders a CardInput with dual branding UI radio button elements', () => {
+        renderCardInput(<CardInput {...cardInputRequiredProps} />);
+
+        void act(() => {
```
Comment (ScottiBR) [inline packages/lib/src/components/Card/components/CardInput/CardInput.dualBrandUI.test.tsx]: When using `void act` means state updates aren't awaited, which can cause flaky tests. add an await similar to other test files

File: packages/lib/src/components/Card/components/CardInput/CardInput.dualBrandUI.test.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        expect(screen.getByText('Card Brand')).toBeInTheDocument();
+
+        expect(screen.getByRole('radio', { name: /VISA/i })).toBeVisible();
+        expect(screen.getByRole('radio', { name: /Carte Bancaire/i })).toBeVisible();
+    });
+
+    test('Dual branding UI is not hidden when the card number is in error', () => {
+        const { container } = renderCardInput(<CardInput {...cardInputRequiredProps} />);
+
+        void act(() => {
```
Comment (ScottiBR) [inline packages/lib/src/components/Card/components/CardInput/CardInput.dualBrandUI.test.tsx]: When using `void act` means state updates aren't awaited, which can cause flaky tests. add an await similar to other test files

File: packages/lib/src/components/Card/components/CardInput/CardInput.dualBrandUI.test.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            cardInputRef.processBinLookupResponse(dualBrandResp, false);
+        });
+
+        expect(screen.getByText('Card Brand')).toBeInTheDocument();
+
+        /* eslint-disable testing-library/no-node-access, testing-library/no-container */
+        // 3 error fields - all hidden
+        expect(container.querySelectorAll('.adyen-checkout-contextual-text--error.adyen-checkout-contextual-text--hidden')).toHaveLength(3);
+
+        void act(() => {
```
Comment (ScottiBR) [inline packages/lib/src/components/Card/components/CardInput/CardInput.dualBrandUI.test.tsx]: When using `void act` means state updates aren't awaited, which can cause flaky tests. add an await similar to other test files

File: packages/lib/src/components/BacsDD/components/BacsInput.test.tsx [hunk 7]
```diff
@@ -1,73 +1,60 @@
...
-        expect(wrapper.find('div.adyen-checkout__bacs--bank-location-id.adyen-checkout__field--inactive')).toHaveLength(1);
-
-        // Email (inactive)
-        expect(wrapper.find('div.adyen-checkout__bacs--shopper-email.adyen-checkout__field--inactive')).toHaveLength(1);
-
-        // No consent checkboxes
-        expect(wrapper.find('ConsentCheckbox')).toHaveLength(0);
+        const { bacsRef } = renderBacsInput({});
+
+        void act(() => {
```
Comment (ScottiBR) [inline packages/lib/src/components/BacsDD/components/BacsInput.test.tsx]: When using `void act` means state updates aren't awaited, which can cause flaky tests. add an await similar to other test files

File: .windsurf/rules/code-guide.md (hunk 2 above)
Comment (ScottiBR) [reply .windsurf/rules/code-guide.md:474]: Maybe you could add the act instructions here as well:
```
- **DO** use `await act()` when triggering state updates via refs or callbacks
- **DO NOT** use `void act()` — always await the returned promise
```
